### DISCUSSION TITLE
how to get offset on scrolling?

### DISCUSSION THREAD

**Discussion by harrybest** [idiotWu/smooth-scrollbar#377] [Q&A] [answered]
i have tried multiple methods but couldn't get it work, console.log(scrollbar.offset); it always showing undefined any code for jquery will be helpful.

**Comment by dolphin-wood** [comment, marked as answer]
`scrollbar.offset` is an instance property, so you need to

```js
const scrollbar = Scrollbar.get(elem /* the element on which the scrollbar is initialized */);
console.log(scrollbar.offset);
```

You can also use the [`addListener()`](https://github.com/idiotWu/smooth-scrollbar/blob/develop/docs/api.md#scrollbaraddlistener) method to get offset on scrolling:

```js
scrollbar.addListener(({ offset }) => {
  console.log(offset);
});

```

**Reply by harrybest** [reply]
Great thanks for quick reply and just find out such things you already provide in document but on previous day was just exhausted and couldn't figure out but today with empty mind things seems quite easy 👍
